frontend/src/components/eMIB/EditActionDialog.jsx

```diff
@@ -119,19 +124,127 @@ class EditActionDialog extends Component {
     this.setState({ action: updatedAction });
   };
 
-  showCancelConfirmationDialog = () => {
-    this.setState({ showCancelConfirmationDialog: true });
+  /* 
+  this handles the display of the cancel confirmation message: Are you sure you want to cancel this response?
+  when creating a new email/task and trying to exit the form, if it's empty, don't display any warning message
+  when creating a new email/task and trying to exit the form, if there is any change to at lest one of the fields, display the cancel warning message
+  when editing an email/task and trying to exit the form, if there is no changes, don't display any warning messsage
+  when editing an email/task and trying to exit the form, if there is any change to at lest one of the fields, display the cancel warning message
+  */
+  handleCancelConfirmationDisplay = () => {
+    const { actionType, editMode, handleClose, action } = this.props;
+
```

Comment:
As a continuation  of the previous comment: would it be possible to add a function call, with the props as parameters, to a sort of "core" function? something like:
... <start above comment>
        handleCancelConfirmationDisplayLogic(actionType, editMode, handleClose, action);
}

export handleCancelConfirmationDisplayLogic(actionType, editMode, handleClose, action){
... <continue below comment>

then move the new function (handleCancelConfirmationDisplayLogic or whatever you call it) Above the "class EditActionDialog extends Component {" (somewhere near the styles),

Then write tests for that?

And completely understand if you want to make that a different PR that gets merged into this one

Reply:
As discussed offline, we'll pair tomorrow morning to work on these new tests.